core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -341,9 +220,13 @@ namespace kagome::primitives {
 
   outcome::result<Buffer> ScaleCodecImpl::encodeExtrinsic(
       const Extrinsic &extrinsic) const {
-    Buffer out;
-    OUTCOME_TRY(scale::TypeEncoder<Extrinsic>{}.encode(extrinsic, out));
-    return out;
+    ScaleEncoderStream s;
+    try {
+      s << extrinsic;
+    } catch (...) {
+      std::terminate();
```

Comment:
Isn't that too hard, taking into account you are returning a result from this function and thus can return an error?

Reply:
this is going to be fixed as mentioned [here](https://github.com/soramitsu/kagome/pull/127#discussion_r287586103)